spec/unit/puppet/provider/registry_key_spec.rb

```diff
@@ -26,4 +44,47 @@
       reg_key.provider.exists?.should be_false
     end
   end
+
+  describe "#purge_values" do
+    let (:guid) { SecureRandom.uuid }
+    let (:reg_path) { "#{puppet_key}\\#{subkey_name}\\Unicode-#{guid}" }
+
+    def bytes_to_utf8(bytes)
+      bytes.pack('c*').force_encoding(Encoding::UTF_8)
+    end
+
+    before(:each) do
+      # create temp registry key with Unicode values
+      Win32::Registry::HKEY_LOCAL_MACHINE.create(reg_path,
+        Win32::Registry::KEY_ALL_ACCESS) do |reg_key|
+          endash = bytes_to_utf8([0xE2, 0x80, 0x93])
+          tm = bytes_to_utf8([0xE2, 0x84, 0xA2])
+
+          # TODO: this doesn't really trigger string encoding issues
+          reg_key.write(endash, Win32::Registry::REG_SZ, tm)
```

Comment:
Encoding::UndefinedConversionError: "\xE2" to UTF-8 in conversion from ASCII-8BIT to UTF-8 to UTF-16LE
  occurred at C:/code/puppetlabs/modules/puppetlabs-registry/.bundle/gems/ruby/1.9.1/gems/puppet-3.7.3-x86-mingw32/lib/puppet/util/windows/string.rb:13:in `encode!'